File: node-js/src/RandomDataGenerator.js
class RandomDataGenerator {
    constructor() {
        this.stockCode = ['AAPL', 'GOOGL', 'MSFT', 'AMZN', 'TSLA'];
        this.stockPrice = [100, 200, 300, 400, 500];
        this.buyOrSell = ['B', 'S'];
    }

    getData() {
        const randomStockCode = this.stockCode[Math.floor(Math.random() * this.stockCode.length)];
        const randomStockPrice = this.stockPrice[Math.floor(Math.random() * this.stockPrice.length)];
        const randomBuyOrSell = this.buyOrSell[Math.floor(Math.random() * this.buyOrSell.length)];
        const randomQuantity = Math.floor(Math.random() * 100) + 1;
        return `${randomStockCode}|${randomStockPrice}|${randomBuyOrSell}|${randomQuantity}`;
    }
}

File: golang/random_data_generator.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

// RandomDataGenerator generates random stock data for simulation
type RandomDataGenerator struct {
	stockCodes  []string
	stockPrices []int
	buyOrSell   []string
	rand        *rand.Rand
}

// NewRandomDataGenerator creates a new instance of RandomDataGenerator with seed
func NewRandomDataGenerator() *RandomDataGenerator {
	source := rand.NewSource(time.Now().UnixNano())
	return &RandomDataGenerator{
		stockCodes:  []string{"AAPL", "GOOGL", "MSFT", "AMZN", "TSLA", "META", "NFLX", "INTC", "AMD", "NVDA"},
		stockPrices: []int{100, 200, 300, 400, 500, 150, 250, 350, 450, 550},
		buyOrSell:   []string{"B", "S"},
		rand:        rand.New(source),
	}
}

// GetData returns a random stock data entry in the format "StockCode|Price|B/S|Amount"
func (g *RandomDataGenerator) GetData() string {
	stockCode := g.stockCodes[g.rand.Intn(len(g.stockCodes))]
	stockPrice := g.stockPrices[g.rand.Intn(len(g.stockPrices))]
	tradeType := g.buyOrSell[g.rand.Intn(len(g.buyOrSell))]
	quantity := g.rand.Intn(100) + 1
	
	return fmt.Sprintf("%s|%d|%s|%d", stockCode, stockPrice, tradeType, quantity)
}
